### PR TITLE
Custom 404 and 500 error pages

### PR DESCRIPTION
**Pull request by LevinGeorge**

### CODE REVIEW

File: app/assets/javascripts/errors.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
```
Comment (danielpaul) [inline app/assets/javascripts/errors.js]: Remove this file?

File: app/assets/stylesheets/errors.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the errors controller here.
```
Comment (danielpaul) [inline app/assets/stylesheets/errors.scss]: Don't need this file too if we don't have any error specific CSS? Even if we do need it, it should be SASS file.

File: app/controllers/errors_controller.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class ErrorsController < ApplicationController
+  def not_found
+    render(:status => 404)
```
Comment (danielpaul) [inline app/controllers/errors_controller.rb]: @LevinGeorge Stick with the new Ruby hash syntax pls :)

File: app/views/errors/internal_server_error.html.haml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+%html
+  %head
+    %title Page not found!
+    %meta{:content => "width=device-width,initial-scale=1", :name => "viewport"}/
+    :css
```
Comment (danielpaul) [inline app/views/errors/internal_server_error.html.haml]: There shouldn't be any inline CSS. Move this to `error.sass` file and include it in the stylesheet. If you want to target the body with CSS, add controller and action name as a class in the body tag in the layouts. Something like `body{class: "c-" + params[:controller] + " a-" + params[:action]}`

File: config/routes.rb [hunk 5]
```diff
@@ -1,5 +1,9 @@
 Rails.application.routes.draw do
 
+  get 'errors/not_found'
```
Comment (danielpaul) [inline config/routes.rb]: Could we keep the routes file organised? Don't add always just to the top. Routes file has to be organised based on priority and matching.

Comment (danielpaul) [inline config/routes.rb]: Oh, just noticed the other routes at the bottom. What are these for?

File: db/schema.rb [hunk 6]
```diff
@@ -25,6 +25,15 @@
     t.datetime "updated_at"
   end
 
+  create_table "comments", force: :cascade do |t|
```
Comment (danielpaul) [inline db/schema.rb]: Why is this changed on this branch?

File: public/404.html [hunk 7]
```diff
@@ -1,67 +0,0 @@
-<!DOCTYPE html>
```
Comment (danielpaul) [inline public/404.html:1]: Keep these files just as a backup that the app does not even start. We need these static ones too.

File: app/controllers/errors_controller.rb (hunk 3 above)
Comment (LevinGeorge) [inline app/controllers/errors_controller.rb]: @danielpaul D'accord :+1:

File: db/schema.rb (hunk 6 above)
Comment (LevinGeorge) [inline db/schema.rb]: I don't know. I didn't even go near the schema

File: config/routes.rb (hunk 5 above)
Comment (LevinGeorge) [inline config/routes.rb]: @danielpaul The match routes make sure that the 404 and 500 files are called for any methods like GET, POST and etc.

File: public/404.html (hunk 7 above)
Comment (LevinGeorge) [inline public/404.html:1]: @danielpaul The reason I deleted these files was because they pop up instead of the dynamic error files when you go to this page 'http://localhost:3000/404'

Comment (LevinGeorge) [inline public/404.html:1]: @danielpaul The other guys also said to delete these public error html files since we already created the dynamic ones.

Comment (danielpaul) [inline public/404.html:1]: Aha okey cool.

File: app/assets/stylesheets/errors.sass [hunk 8]
```diff
@@ -0,0 +1,20 @@
+body#errors
+  background-color: #F4F4F4
+  color: #2E2F30
+  text-align: center
+  font-family: arial, sans-serif
+  margin: 0
+
+h1.errors
```
Comment (danielpaul) [inline app/assets/stylesheets/errors.sass]: These need to be indented so they are inside `body#errors`. Then you don't have to add `.errors` class to `h1` and `p`.

Comment (LevinGeorge) [inline app/assets/stylesheets/errors.sass]: oh ok :+1:

File: app/views/errors/internal_server_error.html.haml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+%html
```
Comment (danielpaul) [inline app/views/errors/internal_server_error.html.haml]: Are we using the same layout as the application layout? If so, we don't need the html tags here. If not, you need to include the partials from the application layout to render the same header and footer.

File: db/schema.rb (hunk 6 above)
Comment (danielpaul) [inline db/schema.rb]: Could we remove these from the schema on this branch? This is not relevant to this branch.

File: app/views/errors/internal_server_error.html.haml (hunk 9 above)
Comment (LevinGeorge) [inline app/views/errors/internal_server_error.html.haml]: Done :)

File: db/schema.rb (hunk 6 above)
Comment (LevinGeorge) [inline db/schema.rb]: Do you want me to remove the comments table from the schema manually on this branch?

Comment (danielpaul) [inline db/schema.rb]: Yes please :)

Comment (danielpaul) [inline db/schema.rb]: Just edit the Schema file and remove the ones that were added to this. Check GitHub file changes on this branch and you can see what was added to the schema file there.

Comment (danielpaul) [inline db/schema.rb]: Just edit the Schema file and remove the ones that were added to this. Check GitHub file changes on this branch and you can see what was added to the schema file there.

File: app/views/errors/internal_server_error.html.haml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+%body#errors_page
```
Comment (danielpaul) [inline app/views/errors/internal_server_error.html.haml]: Is the body not included in the layout too?

File: app/views/errors/internal_server_error.html.haml [hunk 11]
```diff
@@ -0,0 +1,6 @@
+%div#errors_page
```
Comment (danielpaul) [inline app/views/errors/internal_server_error.html.haml:1]: No need to declare `div` here. It's a `div` by default. But it's fine. Just for future ref :P ;)

Comment (LevinGeorge) [inline app/views/errors/internal_server_error.html.haml:1]: :P :P :P :P :P @danielpaul
